components/SmartAnchor.tsx

```diff
@@ -66,6 +73,27 @@ const SmartAnchor: React.FunctionComponent<Props> = ({linkComponent, linkRelativ
   return <Link {...rest} to={destination} />
 };
 
-export function makeSmartAnchor(linkComponent: React.ComponentType<LinkProps>, linkRelative: boolean): React.ComponentType<Props> {
-  return (props: Props) => <SmartAnchor {...props} linkComponent={linkComponent} linkRelative={linkRelative} />
+const SmartAnchorContext = React.createContext<React.ComponentType<BaseProps> | undefined>(undefined);
+
+export const WithSmartAnchor: React.FunctionComponent<{
+  linkComponent: React.ComponentType<LinkProps>;
+  linkRelative: boolean;
+}> = ({linkComponent, linkRelative, children}) => {
+  return (
+    <SmartAnchorContext.Provider value={makeSmartAnchor(linkComponent, linkRelative)}>
+      {children}
+    </SmartAnchorContext.Provider>
+  )
+}
+
+export function useSmartAnchor(): React.ComponentType<BaseProps> {
```

Comment:
With MDX, we can use shortcodes to inject new component types globally, but since check documentation is plain React, not MDX, we need to use this injection mechanism (we're doing something similar for icons).